src/components/TorrentDetail/Overview.vue

```diff
@@ -152,12 +156,24 @@ function handleKeyboardShortcuts(e: KeyboardEvent) {
   }
 }
 
-onMounted(() => {
+onMounted(async () => {
   document.addEventListener('keydown', handleKeyboardShortcuts)
+  if (canvas.value) {
+    canvas.value.width = 4096
+    if (piecesApp === null) {
+      const app = new Application()
+      await app.init({ antialias: true, width: canvas.value.width, height: canvas.value.height })
+      for (const attr of [...canvas.value.attributes]) app.canvas.setAttribute(attr.nodeName, attr.nodeValue ?? '')
+      canvas.value.replaceWith(app.canvas)
+      piecesApp = app
+    }
+  }
```

Comment:
@Larsluph I've just pushed changes that wait for refs to load properly, fixes everything mentioned here.

Reply:
> Do you know of the correct way to wait for `canvas.value` to become available?

You can use a watcher on the canvas' ref to do that but I think the onMounted waits for the DOM to load the entities. The canvas may not be loaded because of the current restrictions which sometimes makes it not render.

Reply:
I was doing console.log at the start of `onMounted` and maybe 1/10 times ref values would be undefined when it fired.  Current PR works 100% now compensating for these delayed loads   Let me know if I should change it.

Reply:
> > Do you know of the correct way to wait for `canvas.value` to become available?
> 
> You can use a watcher on the canvas' ref to do that but I think the onMounted waits for the DOM to load the entities. The canvas may not be loaded because of the current restrictions which sometimes makes it not render.

Both `canvas.value` and `shouldRefreshPieceState.value` sometimes load as undefined, I don't think `onMounted` guarantees they will be loaded.

Reply:
> I don't think `onMounted` guarantees they will be loaded.

As per [the docs](https://vuejs.org/api/composition-api-lifecycle.html#onmounted), you should be able to access DOM elements in this hook... That's weird 🤔

Reply:
I think the DOM is indeed ready, but from what I'm reading refs sometimes take at least one or two ticks to update.

https://jefrydco.id/en/blog/safe-access-vue-refs-undefined#conclusion

Reply:
https://vuejs.org/guide/essentials/reactivity-fundamentals.html#dom-update-timing

Reply:
@Larsluph 

Just in case, I tested even using `nextTick()` as recommended in Vue docs, that didn't guarantee success:

test code:
```javascript
onMounted(() => {
  document.addEventListener('keydown', handleKeyboardShortcuts)
  console.log('before-promise', document.querySelector( 'canvas' ), canvas.value, shouldRefreshPieceState.value)
  nextTick( () => 
  piecesApp = new Promise<Application | false>(async function waitForPiecesAppPromise(resolve, reject) {
    console.log('in-promise', document.querySelector( 'canvas' ), canvas.value, shouldRefreshPieceState.value)
    await nextTick()
    console.log('after-2nd-nexttick', document.querySelector( 'canvas' ), canvas.value, shouldRefreshPieceState.value)
    if (!canvas.value) {
      reject(new Error('canvas.value failed to initialize in time'))
      return
    }
```

log when succeeds:
```
TorrentDetail-L3-NQMbP.js:12 before-promise <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ true
12:05:09.477 TorrentDetail-L3-NQMbP.js:12 in-promise <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ true
12:05:09.478 TorrentDetail-L3-NQMbP.js:12 after-2nd-nexttick <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ <canvas data-v-3030a4d4 width=​"4096" height=​"10">​ true
```

log when fails:
```
TorrentDetail-L3-NQMbP.js:12 before-promise null undefined false
12:07:17.592 TorrentDetail-L3-NQMbP.js:12 in-promise null undefined false
12:07:17.593 TorrentDetail-L3-NQMbP.js:12 after-2nd-nexttick null undefined false
12:07:17.606 TorrentDetail-L3-NQMbP.js:12 Uncaught (in promise) Error: canvas.value failed to initialize in time
    at TorrentDetail-L3-NQMbP.js:12:167244
```

I hit F5 (refresh) on the browser 12 times and got the following results:
succeed, succeed, succeed, fail, succeed, succeed, succeed, fail, succeed, fail, succeed, fail

I'm not sure if there is a non-polling solution to ensure proper instantiation?

Reply:
Unfortunately having not worked with Vue before I don't know how to get there.  Do you think that this PR will work as is?  You guys could always further adapt it later.

Reply:
I will try that later tonight

Reply:
I'll let you check that everything is good. My canvas is full black when testing to render it, it seems that I broke something while moving things around... 🤔

Reply:
> I'll let you check that everything is good. My canvas is full black when trying to render it, it seems that I broke something while moving things around... 🤔

I wasn't Immediately able to figure out why it isn't working either.  I have to go out for a bit but I'll check later when I can.

Reply:
I did update the canvas width to 4096, this gives maximum details while still maintaining compatibility with nearly all clients.

Reply:
@Larsluph 

Figured it out, pushing a fix soon, app doesn't like to be deeply reactive with `ref`, have to use `shallowRef`.

Also not the cause but `app` needs to be saved as a promise so `renderCanvas` can wait for `init` to finish before trying to access it.  Currently execution can happen like this with `graphics` being added before `init` finishes:
```
22:02:15.530 TorrentDetail-GfvQE6Dx.js:12749 app pre-init
22:02:15.536 TorrentDetail-GfvQE6Dx.js:12705 renderCanvas
22:02:15.850 TorrentDetail-GfvQE6Dx.js:12736 addChild graphics
22:02:15.995 TorrentDetail-GfvQE6Dx.js:12756 app post-init
```

Reply:
Fixes pushed in https://github.com/VueTorrent/VueTorrent/pull/1564